shell/browser/mac/in_app_purchase.mm

```diff
@@ -48,11 +48,11 @@ @implementation InAppPurchase
  */
 - (id)initWithCallback:(in_app_purchase::InAppPurchaseCallback)callback
               quantity:(NSInteger)quantity
-              username:(NSString)username {
+              username:(NSString*)username {
   if ((self = [super init])) {
     callback_ = std::move(callback);
     quantity_ = quantity;
-    username_ = username;
+    username_ = [username copy];
```

Comment:
if we are copying this, do we need to make sure we release it? is this leaking as is?

Reply:
Good point.